onnxruntime/contrib_ops/cuda/sparse/sparse_attention_helper.h

```diff
@@ -202,13 +202,13 @@ Status CheckInputs(void* params,
                              "head_size shall be a multiple of 16. Got head_size = ",
                              head_size);
     }
-    if (cos_dims[0] < max_sequence_length) {
+    if (cos_dims[0] < total_sequence_length) {
```

Comment:
Is the fix hit any tests under onnxruntime? If not, could you add a unit test for it?

Reply:
We can add it in [this PR](https://github.com/microsoft/onnxruntime/pull/20614).